mesonbuild/serialisation.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2013-2021 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing as T
+
+if T.TYPE_CHECKING:
+    from . import build
+    from . import programs
+    from .backend.backends import TestProtocol
+
+class ExecutableSerialisation:
+
+    # XXX: should capture and feed default to False, instead of None?
+
+    def __init__(self, cmd_args: T.List[str],
+                 env: T.Optional['build.EnvironmentVariables'] = None,
+                 exe_wrapper: T.Optional['programs.ExternalProgram'] = None,
+                 workdir: T.Optional[str] = None,
+                 extra_paths: T.Optional[T.List] = None,
+                 capture: T.Optional[bool] = None,
+                 feed: T.Optional[bool] = None,
+                 tag: T.Optional[str] = None,
+                 verbose: bool = False,
+                 ) -> None:
+        self.cmd_args = cmd_args
+        self.env = env
+        if exe_wrapper is not None:
+            # XXX: import is a bit slow - but perhaps mypy type checking obsoleted this
+            # anyway?
+            from . import programs
```

Comment:
^

Reply:
The next line does assert isinstance(exe_wrapper, programs.ExternalProgram), which is the only reason ExternalProgram is needed at that point. I was wondering if that assert isn't really needed anymore, due to the mypy type checking.

Reply:
Yes, I believe mypy renders this unneeded.

Reply:
`./run_mypy.py` pass locally without that assert, so I think you can drop it.

Reply:
Asserts were added in 416a00308f5b0f228af3c93eb597eca8529fdbb0, and the file got properly typed in f6f35aa906c8ebac21af07c17449df65d08aeaeb.